pinot-core/src/main/java/org/apache/pinot/core/common/datatable/DataTableImplV4.java

```diff
@@ -48,7 +50,26 @@ public DataTableImplV4(int numRows, DataSchema dataSchema, Map<String, Map<Integ
   }
 
   @Override
-  protected int getDataBlockVersionType() {
+  public RoaringBitmap getNullRowIds(int colId) {
+    // _fixedSizeData stores two ints per col's null bitmap: offset, and length.
+    int position = _numRows * _rowSizeInBytes + colId * Integer.BYTES * 2;
+    _fixedSizeData.position(position);
```

Comment:
We should check the length of the buffer. If `nullRowIds` are not passed, fixed size data length will be smaller than the position, and we should return `null` in that scenario